index.bs

```diff
@@ -180,88 +181,180 @@ public IP address scope is being discussed in
-    [=device=] becomes
-    [potentially trustworthy](https://w3c.github.io/webappsec-secure-contexts/#potentially-trustworthy-origin)
-    in a certain period:
-    - a [=pre-installed certificate=] granted on the web application’s origin
-	- a certificate issued by the [=device CA=]
-- If the web server allows the UA to grant the [=pre-installed certificate=] of
-    the [=device=] in its web application’s origin, the web server shall be able
-    to provide the UA with a hint about the certificate to be granted.
-- If the [=device=] requires a certificate issued by the [=device CA=], the
-    [=device=] shall request the [=device CA=] to issue a certificate for the
-    [=device=] when the user grants.
-- The [=device CA=] shall verify certificate issuing request from both the
-    [=device=] and the UA before issuing a certificate to the [=device=].
-- The certificate shall have a reasonably short expiration period.
-
-### <dfn>Certificate Management</dfn> ### {#req-certificate-management}
-
-- The UA shall ask the user if the user grants the renewal of the certificate
-    when the certificate is expired during [=device identification=],
-    [=certificate grant and issuance=], or communication with the [=device=].
-- The UA shall be able to revoke the certificate when the user decides to do so.
-- The certificate issued by the [=device CA=] shall be able to be revoked by the
-    [=device CA=] when necessary.
-
-## Non-Functional Requirements ## {#req-non-funcational-requirements}
-
-### Privacy and Security ### {#req-privacy-and-security}
-
-- [=Device identification=] and [=certificate grant and issuance=] should
-    prevent passive network eavesdroppers from learning messages related to
-    identification or certificate passed between the UA and the device.
-- [=Device identification=] and [=certificate grant and issuance=] should
-    prevent active network attackers from impersonating a device and observing
-    or altering data intended for the UA.
+In case that the user inputs the device URL to the address bar of the [=UA=] directly ([[#uc-07]])
+or the [=UA=] shows the popup window to the user to get the user's approval ([[#uc-02]], [[#uc-03]]),
+the user may be able to recognize that the device URL displayed or input on the [=UA=]
+indicates the real device on the local network. In such cases, the [=device=] and/or
+the [=UA=] shall meet the following requirements:
+- The device identifier (domain name) should be easy to remember and short enough but
+    not to recognize as fraudulent by anti-phishing.
+- The [=device=] and the [=UA=] should have a pairing method by using the device
+    interface (e.g., pushing a specific button on the [=device=] to start the pairing,
+    showing the PIN code for the pairing on the [=device=]’s display).
+
+### Alerting User and User's Approval ### {#req-alerting-user}
+
+The [=web service=] loaded on the [=UA=] should not be able to access the [=device=]
```

Comment:
"The web service *loaded on the UA*" looks a little bit strange, because a "web service" is defined above as "hosted on the internet" (i.e. running on a web server).

Reply:
Okay. It's not correct. I should have written it like this: "The frontend page of the web service (loaded on the UA) should not be able to ..."